src/Routing/Getter.purs

```diff
@@ -0,0 +1,123 @@
+module Routing.Getter (
+  route,
+  or,
+  contains,
+  runRoute,
+  hashes,
+  checks,
+  routes,
+  PErr(),
+  Route(),
+  RouteMsg,
+  toMsg,
+  Check(),
+  Checks()
+  ) where
+
+import Data.Either
+import Data.Maybe
+import Data.Tuple
+import Control.Monad.Eff
+import Control.Monad.State
+import Control.Monad.Trans
+import Control.Monad.State.Class
+import Control.Monad.Error
+import Control.Apply ((*>))
+import Control.Alt ((<|>))
+import Data.String.Regex (replace, noFlags, Regex(), regex)
+import Data.Array (findIndex, drop, take)
+import Data.Traversable (traverse)
+
+import qualified Data.StrMap as M
+import qualified Text.Parsing.Parser as P
+import qualified Text.Parsing.Parser.Combinators as P
+
+import Routing.Parser (parse, template, StateObj())
+
+type RouteParser = P.ParserT String StateObj Checks
+type Check = Tuple String (M.StrMap String)
+type Checks = Tuple Check [Check]
```

Comment:
Where does the name `Check` come from here? I think something like `RouteMatch` would be more intuitive.